### PR TITLE
Factory girl integration

### PR DESCRIPTION
**Pull request by ncrawford0**

### CODE REVIEW

File: spec/features/reviews/user_edits_review_spec.rb [hunk 1]
```diff
@@ -1,39 +1,45 @@
...
-    fill_in "Title", with: "Best bar I've been to"
-    fill_in "Review", with: "We were visiting and had a good time"
-    fill_in "Rating", with: 5
-    click_button "Create Review"
-  end
+  let!(:user1) { FactoryGirl.create(:user) }
+  let!(:user2) { FactoryGirl.create(:user, email: "janedoe@gmail.com") }
+  let!(:bar) { FactoryGirl.create(:bar, user: user1) }
+  let!(:review1) { FactoryGirl.create(:review, user: user1, bar: bar) }
+  let(:review2) { FactoryGirl.create(:review, user: user1, bar: bar, title: "Worst bar ever.", body: "My freinds and I had a horible time here.", rating: 1) }
```
Comment (kevlarr) [inline spec/features/reviews/user_edits_review_spec.rb]: This really is way too long, it should be converted to a multiline:

```
let(:review2) do
  FactoryGirl.create(:review,
    user: user1,
    bar: bar,
    title: "Worst bar ever.",
    body: "My freinds and I had a horible time here.",
    rating: 1
  )
end
```

File: spec/features/reviews/user_deletes_review_spec.rb [hunk 2]
```diff
@@ -1,43 +1,42 @@
 require "rails_helper"
 
 feature "user deletes a review" do
-  let!(:user1) { User.create(email: "mckelvey.matt@gmail.com", password: "12345678") }
-  let!(:user2) { User.create(email: "abcd4@gmail.com", password: "12345678") }
+  let!(:user1) { FactoryGirl.create(:user) }
+  let!(:user2) { FactoryGirl.create(:user, email: "janedoe@gmail.com") }
+  let!(:bar) { FactoryGirl.create(:bar, user: user1) }
+  let!(:review1) { FactoryGirl.create(:review, user: user1, bar: bar) }
+  let(:review2) { FactoryGirl.create(:review, user: user1, bar: bar, title: "Worst bar ever.", body: "My freinds and I had a horible time here.", rating: 1) }
```
Comment (kevlarr) [inline spec/features/reviews/user_deletes_review_spec.rb]: Make this multiline

File: spec/factories/reviews.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :review do
+    title "Great Atmosphere"
+    body "Me and my partner had a good evening here with some firends.  The food was great!"
```
Comment (kevlarr) [inline spec/factories/reviews.rb:4]: Ignore this

File: spec/features/bars/user_edits_bar_spec.rb [hunk 4]
```diff
@@ -1,29 +1,31 @@
 require "rails_helper"
 
 feature "user edits an existing bar" do
-  let!(:user1) { User.create(email: "abcd2@gmail.com", password: "12345678") }
-  let!(:user2) { User.create(email: "abcd4@gmail.com", password: "12345678") }
-  let!(:bar1) { Bar.create(name: "Happy Hour Lasagna", description: "Where are we at?", user: user1) }
+  let!(:user1) { FactoryGirl.create(:user) }
+  let!(:user2) { FactoryGirl.create(:user, email: "janedoe@gmail.com") }
+  let!(:bar1) { FactoryGirl.create(:bar, user: user1) }
+  let(:bar2) { FactoryGirl.create(:bar, name: "The Yard House", description: "A brewhouse near fenway where you can buy yards of beer.", user: user1) }
```
Comment (kevlarr) [inline spec/features/bars/user_edits_bar_spec.rb]: Make this multiline

File: spec/features/reviews/user_deletes_review_spec.rb [hunk 5]
```diff
@@ -1,43 +1,42 @@
...
+    visit bars_path
+    click_button "Sign in"
+    fill_in "Email", with: user2.email
+    fill_in "Password", with: user2.password
+    click_button "Log in"
+    click_link bar.name
+    click_button "Edit"
+
+    expect(page).to have_content bar.name
+    expect(page).to have_content("You do not have permission to make this change")
```
Comment (kevlarr) [inline spec/features/reviews/user_deletes_review_spec.rb:26]: Ignore this

File: spec/factories/bars.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :bar do
+    name "Game On"
```
Comment (justinhuynh) [inline spec/factories/bars.rb:3]: You might want to sequence the name so they're unique. `sequence(:name) { |n| "Game On #{n}" }`

**Pull request merged**